dask/dataframe/io/io.py

```diff
@@ -588,10 +593,12 @@ def from_delayed(
         if not isinstance(df, Delayed):
             raise TypeError("Expected Delayed object, got %s" % type(df).__name__)
 
+    parent_meta = delayed(make_meta_util)(dfs[0]).compute()
+
     if meta is None:
-        meta = delayed(make_meta)(dfs[0]).compute()
+        meta = parent_meta
     else:
-        meta = make_meta(meta)
+        meta = make_meta_util(meta, parent_meta=parent_meta)
```

Comment:
Is there a good reason to pass `parent_meta` if a `meta` object is already provided? That is, shouldn't `meta` already be an appropriate DataFrame type?  My intuition tells me that the above changes should just be swapping `make_meta` with `make_meta_util`.

Reply:
> That is, shouldn't `meta` already be an appropriate DataFrame type? 

Sadly, nope. For example:

```python
> /nvme/0/pgali/cudf/dask/dask/dataframe/io/io.py(603)from_delayed()
(Pdb) meta
[('a', 'f8'), ('b', 'f8'), ('c', 'f8'), ('d', 'f8')]
(Pdb) parent_meta
Empty DataFrame
Columns: [a, b, c, d]
Index: []
```